### PR TITLE
Add check to catch ytt rendering errors

### PR DESCRIPTION
**Pull request by moleske**
Co-authored-by: Michael Oleske <moleske@pivotal.io>
Co-authored-by: Dave Walter <dwalter@pivotal.io>

### CODE REVIEW

File: matchers/produce_yaml_matcher.go [hunk 1]
```diff
@@ -34,6 +36,12 @@ func (matcher *ProduceYAMLMatcher) Match(actual interface{}) (bool, error) {
 	}
 
 	matcher.rendered = string(session.Out.Contents())
+
+	if session.ExitCode() != 0 {
```
Comment (paulcwarren) [inline matchers/produce_yaml_matcher.go:40]: Seems like this error is already be handled and returned on (new) line 34 just above.  I cant think how we would ever enter this if block.

Comment (Birdrock) [reply matchers/produce_yaml_matcher.go:40]: I agree. The difference appears to mostly involve error messaging.

Comment (davewalter) [reply matchers/produce_yaml_matcher.go:40]: I would be curious to see what a ytt rendering error looks like with our current error. It seems like a multi-line error inside of `command={%s}` would not be ideal. It also strikes me as not great Golang practice to look at the other parameter from a function when the error is not nil. IMHO, we really should split this check into separate pieces; one for `err != nil` and the other for `session.ExitCode() != 0`.

Comment (DennisDenuto) [reply matchers/produce_yaml_matcher.go:40]: This looks to us like unreachable code. We are going to remove this if statement and check to see how it affects tas-for-k8s unit tests that ran with that seemingly unreachable code.

@davewalter + @Birdrock  as for the error message, we will update this thread with what the assertion error message looks like when session returns a non-zero exit code (with the 'unreachable code' removed).

Comment (paulcwarren) [reply matchers/produce_yaml_matcher.go:40]: Thanks @DennisDenuto I have been trying to find to do this but been unable so this is very much appreciated.

Comment (DennisDenuto) [reply matchers/produce_yaml_matcher.go:40]: no worries @paulcwarren! 

tas for k8s would like to make an assertion on ytt rendering errors. specifically we want to make an assertion that looks like this:

```
Expect(ctx).To(ProduceYAML(MatchError(ContainSubstring("assert.fail: fail: missing required property [usage_service.database.host]"))))
```

The 'unreachable code' allows a matcher to be 'matched' against the standard error returned by ytt.

However it is unreachable! we would like to run this unreachable code (so we can make assertions on errors). Therefore @davewalter suggestion of splitting ` err != nil` and  `session.ExitCode() != 0` makes sense for us.

We are going to update this PR with the above change.

**Comment by DennisDenuto** [comment]
Closing this PR in favor of https://github.com/cloudfoundry/yttk8smatchers/pull/5

**Pull request closed without merging**